### PR TITLE
[alertmanager] Adding hostAliases to Alertmanager chart.

### PR DESCRIPTION
**Pull request by DandyDeveloper**
@monotek @naseemkullah 

#### What this PR does / why we need it
Adding hostAlises support to the Alertmanager chart. This is a necessary requirement for my organization due to some interesting network decisions. 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by naseemkullah — APPROVED** [review]

**Comment by DandyDeveloper** [comment]
@naseemkullah Looks like we need some approval to run workflows as well?

File: charts/alertmanager/values.yaml [hunk 1]
```diff
@@ -51,6 +51,15 @@ dnsConfig: {}
   #   - name: ndots
   #     value: "2"
   #   - name: edns0
+hostAliases: {}
```
Comment (monotek) [inline charts/alertmanager/values.yaml:54]:
```suggestion

hostAliases: {}
```

File: charts/alertmanager/Chart.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ icon: https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/a
 sources:
   - https://github.com/prometheus/alertmanager
 type: application
-version: 0.27.0
+version: 0.27.1
```
Comment (monotek) [inline charts/alertmanager/Chart.yaml:9]: new feature should raise minor version

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by DandyDeveloper** [comment]
@naseemkullah Seems I need re-approval :)

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**